seqr/views/apis/data_manager_api_tests.py

```diff
@@ -763,76 +788,85 @@ def _set_file_iter_stdout(rows):
                     new_sample_guid = self._check_rna_sample_model(
-                        individual_id=individual_id, data_source='new_muscle_samples.tsv.gz',
+                        individual_id=individual_tuple[0], data_source='new_muscle_samples.tsv.gz',
                         tissue_type=params.get('created_sample_tissue_type'),
+                        project_name=individual_tuple[1],
                     )
                     self.assertTrue(new_sample_guid in response_json['sampleGuids'])
-                    info_log_calls = [mock.call(info_log, self.data_manager_user) for info_log in info]
-                    if test_round == 0:
-                        info_log_calls.insert(1, mock.call(
-                            'create 1 Samples', self.data_manager_user, db_update={
-                                'dbEntity': 'Sample', 'entityIds': [response_json['sampleGuids'][sample_guid_idx]],
-                                'updateType': 'bulk_create',
-                            }
-                        ))
-                    mock_logger.info.assert_has_calls(info_log_calls)
+                    mock_logger.info.assert_has_calls([mock.call(info_log, self.data_manager_user) for info_log in info])
+                    mock_model_logger.info.assert_has_calls([mock.call(
+                        f'create {num_created_samples} Samples', self.data_manager_user, db_update={
+                            'dbEntity': 'Sample',
+                            'entityIds': response_json['sampleGuids'] if num_created_samples > 1 else\
+                                [response_json['sampleGuids'][-1]],
+                            'updateType': 'bulk_create',
+                        }
+                    )])
 
                     return response_json, new_sample_guid
 
                 # Test loading new data
                 mock_open.reset_mock()
-                mock_logger.reset_mock()
                 mock_load_uploaded_file.return_value = [['NA19675_D2', 'NA19675_1']]
                 mock_writes = []
                 def mock_write(content):
                     mock_writes.append(content)
                 mock_open.return_value.__enter__.return_value.write.side_effect = mock_write
                 body.update({'ignoreExtraSamples': True, 'mappingFile': {'uploadedFileId': 'map.tsv'}, 'file': RNA_FILE_ID})
                 response_json, new_sample_guid = _test_basic_data_loading(
-                    params['new_data'], params["num_parsed_samples"], 2,
-                    '1kg project nåme with uniçøde, Test Reprocessed Project', 2, 16, 1)
-                self.assertTrue(RNA_SAMPLE_GUID in response_json['sampleGuids'])
-                warnings = [f'Skipped loading for the following {len(params["skipped_samples"].split(","))} '
-                            f'unmatched samples: {params["skipped_samples"]}']
-                if params.get('extra_warnings'):
-                    warnings = params['extra_warnings'] + warnings
```

Comment:
it seems like the previous behvaior of specifying the skipped samples and the extra_warnings would still work. Is there a reason you changed this?

Reply:
The `extra_warnings` are generated when the data mismatch tissue types with the existing sample. We want to create new samples for that case; no data will be skipped, so no warning is required.

Reply:
that makes sense. Why change `skipped_samples` to `earnings` though?

Reply:
We could use `skipped_samples` to generate the warning messages. But when there are different numbers of warning messages, it is getting harder to generate them. Moving the warning messages into the `params` works better than generating them.

Reply:
I really just don't understand why you made this change at all though. Before, Outlier had 1 warning about skipped data generated from skipped_samples and TPM had 2 warnings, one generated by skipped_samples and 1 in extra_warnings. We still have that exact same pattern of warnings, why not just change the value of extra_warnings for the warning that changed instead of completely redoing this part of the code?